azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -176,6 +181,8 @@ def initialize(self):
                     log_cgroup_warning("Unable to determine which cgroup version to use: {0}".format(ustr(e)), send_event=True)
                     return
 
+                self.__setup_azure_slice()
```

Comment:
We need to setup the slices before checking for v2 because we need to create the log collector slice.

Reply:
let's add this as a comment in the code